torch/_refs/__init__.py

```diff
@@ -4104,21 +4105,18 @@ def linspace(
     pin_memory: bool = False,
     requires_grad: bool = False,
 ) -> TensorLikeType:
-    if dtype is None:
-        dtype = torch.get_default_dtype()
-
-    # NB: NumPy actually doesn't do this cast, but for this ref, I'd rather have this
-    #     cast than not, because it allows us to always go into the precise path
-    #     if dtype is integral and not worry about whether start/end are float
-    if prims.utils.is_integer_dtype(dtype):
-        if isinstance(start, FloatLike):
-            start = sym_int(start)
-        if isinstance(end, FloatLike):
-            end = sym_int(end)
-
+    result_dtype = utils.type_to_dtype(utils.get_higher_type(type(start), type(end)))
     if py_any(isinstance(arg, complex) for arg in (start, end, steps)):
-        raise NotImplementedError
-    assert not isinstance(start, complex) and not isinstance(end, complex)  # for mypy
+        if dtype is None:
```

Comment:
Why not use

https://github.com/pytorch/pytorch/blob/9f11ce7f67612d1c11f1a6a9b264779b27062e82/torch/_prims_common/__init__.py#L1006

?

Reply:
I just reproduced the checks in core really. I'd vote to leave it as it is as here we'd need to perform casting between 4 different dtypes (start, end, step and dtype).